Fakebook.Posts/Fakebook.Posts.RestApi/Controllers/PostsController.cs

```diff
@@ -358,22 +358,23 @@ public async Task<ActionResult> UploadPicture(IFormFile file, string userId)
         /// <br/>200 OK on success
         /// </returns>
         [Authorize]
-        [HttpGet("newsfeed")]
+        [HttpPost("newsfeed")]
         [ProducesResponseType(StatusCodes.Status200OK)]
-        public async Task<IActionResult> GetNewsfeedAsync()
+        public async Task<IActionResult> GetNewsfeedAsync(NewsFeedDto newsfeedemails)
         {
-            var email = User.FindFirst(ct => ct.Type.Contains("nameidentifier")).Value;
-            var newsfeedPosts = await _postsRepository.NewsfeedAsync(email, 3);
-            // var followedUserEmails = _followsRepository.GetFollowedEmails(email);
-            // followedUserEmails.Add(email);
-            // // TODO: This query MUST be tested as EF may may not be able to convert it to sql!
-            // // In case it doesn't work the posts repo will use the sql in NewsfeedAsync.
-            // var newsfeedPosts = await _postsRepository.AsQueryable()
-            // .Where(p => followedUserEmails.Contains(p.UserEmail))
-            // .GroupBy(p => p.UserEmail)
-            // .SelectMany(g => g.OrderByDescending(p => p.CreatedAt).Take(3))
-            // .ToListAsync();
-            return Ok(newsfeedPosts);
+            if (newsfeedemails != null)
+            {
+                var email = User.FindFirst(ct => ct.Type.Contains("nameidentifier")).Value;
+                newsfeedemails.Emails.Add(email);
+
+
+                var newsfeedPosts = await _postsRepository.NewsfeedAsync(newsfeedemails.Emails);
+                return Ok(newsfeedPosts);
+            }
+            else
+            {
+                throw new ArgumentNullException(nameof(newsfeedemails), "The NewsFeedDto cannot be null.");
```

Comment:
it's never going to be null, model binding will always instantiate it with default values even if there is no request data.

Reply:
I think Damion added that because SonarCloud was giving him a code smell telling him to validate the argument. I guess it should probably be dismissed then.

Reply:
erased.